### PR TITLE
resolve tradeLog.WalletAddress from tx on no WalletFee trade

### PR DESCRIPTION
**Pull request by secmask**
- replace use WalletFee with tradeLog.WalletAddress in approxiate case
- retry to resolve tradeLog.WalletAddress from tx on no WalletFee trade

### CODE REVIEW

File: tradelogs/crawler_v3.go [hunk 1]
```diff
@@ -74,6 +120,26 @@ func (crawler *Crawler) assembleTradeLogsV3(eventLogs []types.Log) ([]common.Tra
 			if err != nil {
 				return nil, errors.Wrapf(err, "failed to get transaction receipt tx: %v", tradeLog.TransactionHash)
 			}
+			if len(tradeLog.WalletFees) == 0 { // in case there's no fee, we try to get wallet addr from tradeWithHint input
```
Comment (ductm54) [inline tradelogs/crawler_v3.go:123]: pls consider merge this one in [updateBasicInfo](https://github.com/KyberNetwork/reserve-stats/blob/develop/tradelogs/crawler.go#L366) and remove set wallet name and address in crawler v1 and v2
As in updateBasicInfo already call getTransactionByHash

File: tradelogs/crawler_v3.go [hunk 2]
```diff
@@ -74,6 +120,26 @@ func (crawler *Crawler) assembleTradeLogsV3(eventLogs []types.Log) ([]common.Tra
...
+				tx, err := crawler.getTransactionByHash(tradeLog.TransactionHash, defaultTimeout)
+				if err != nil {
+					return nil, errors.Wrapf(err, "failed to get transactionByHash tx: %v", tradeLog.TransactionHash)
+				}
+				if bytes.Equal(tx.To().Bytes(), crawler.networkProxy.Bytes()) { // try to fail early, tx must have dst == networkProxy
+					tradeParam, err := decodeTradeWithHintParam(tx.Data())
+					if err == nil {
```
Comment (ductm54) [inline tradelogs/crawler_v3.go:130]: should we handle edge case 1st

File: tradelogs/crawler_v2.go [hunk 3]
```diff
@@ -53,6 +53,16 @@ func (crawler *Crawler) getTransactionReceipt(txHash ethereum.Hash, timeout time
 	return receipt, nil
 }
 
+func (crawler *Crawler) getTransactionByHash(txHash ethereum.Hash, timeout time.Duration) (*types.Transaction, error) {
```
Comment (ductm54) [inline tradelogs/crawler_v2.go:56]: this one does not  use anywhere

File: tradelogs/cmd/trade-logs-crawler/main.go [hunk 4]
```diff
@@ -215,15 +225,14 @@ func run(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
-
+	networkProxyAddr := common2.HexToAddress(c.String(networkProxyAddrFlag))
```
Comment (halink0803) [inline tradelogs/cmd/trade-logs-crawler/main.go:228]: I think we already have the proxyContracts address here at: https://github.com/KyberNetwork/reserve-stats/blob/8fe023b2a259b0534cc15301641f23dbe17111a2/lib/contracts/addresses.go#L70.

Flag bring the flexible but I think it is redundant here.

**Pull request merged**